src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/CollectMultipleMetricsSparkIntegrationTest.java

```diff
@@ -0,0 +1,131 @@
+
+    //TODO: we're reaching over into picard for these test files
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "picard/analysis/CollectInsertSizeMetrics");
+
+    public String getTestedClassName() {
+        return CollectMultipleMetricsSpark.class.getSimpleName();
+    }
+
+    @DataProvider(name="metricsTestFiles")
+    public Object[][] insertSizeMetricsFiles() {
+        return new Object[][] {
+                {"insert_size_metrics_test.bam", null},
+                {"insert_size_metrics_test.cram", hg19_chr1_1M_Reference}
+        };
+    }
+
+    @Test(dataProvider="metricsfiles", groups = "spark")
+    public void test(final String fileName, final String referenceName) throws IOException {
+
+        // set up test data input and result outputs (two: one text one histogram plot in pdf)
+        final File input = new File(TEST_DATA_DIR, fileName);
+
+        // create a directory to contain the results since there will be multiple collectors
+        // and each may create multiple files
+        final File outDir = BaseTest.createTempDir("collectMultiMetricsTest");
+        String outBase = outDir.getAbsolutePath() + "/collectMultiMetrics";
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+
+        // IO arguments
+        args.add("-" + StandardArgumentDefinitions.INPUT_SHORT_NAME);
+        args.add(input.getAbsolutePath());
+
+        // The output arg specifies only the basename from which output file(s)
+        // will derived
+        args.add("-" + StandardArgumentDefinitions.OUTPUT_SHORT_NAME);
+        args.add(outBase);
+
+        if (null != referenceName) {
+            final File REF = new File(referenceName);
+            args.add("-" + StandardArgumentDefinitions.REFERENCE_SHORT_NAME);
+            args.add(REF.getAbsolutePath());
+        }
+
+        // for now, run the only two conforming collectors that we have
+        args.add("--collectors");
+        args.add("CollectInsertSizeMetrics");
+
+        args.add("--collectors");
+        args.add("CollectQualityYieldMetrics");
+
+        this.runCommandLine(args.getArgsArray());
+
+        validateInsertSizeMetrics(outBase);
+        validateQualityYieldMetrics(outBase);
+    }
+
+    private void validateQualityYieldMetrics(final String outBase) throws IOException {
+        String localOutBase = outBase + ".qualityYieldMetrics" + ".txt";
+        final MetricsFile<InsertSizeMetrics, Comparable<?>> output = new MetricsFile<>();
+
+        try (FileReader reader = new FileReader(localOutBase)) {
+            output.read(reader);
+            Assert.assertEquals(output.getMetrics().size(), 1);
+        }
+    }
+
+    private void validateInsertSizeMetrics(final String outBase) throws IOException {
+        String localOutBase = outBase + ".insertMetrics" + ".txt";
+        final MetricsFile<InsertSizeMetrics, Comparable<?>> output = new MetricsFile<>();
+
+        try (FileReader reader = new FileReader(localOutBase)) {
```

Comment:
\[FindBugs\] WARNING: Dm: Found reliance on default encoding in org\.broadinstitute\.hellbender\.tools\.spark\.pipelines\.metrics\.CollectMultipleMetricsSparkIntegrationTest\.validateInsertSizeMetrics\(String\): new java\.io\.FileReader\(String\)